File: Java/HashMap/N-Repeated Element in Size 2N Array.java
class Solution {
    public int repeatedNTimes(int[] nums) {
        int sz = nums.length;
        int n = sz/2;
        HashMap<Integer,Integer> mp = new HashMap<>();
        for(int num:nums){
            mp.put(num,mp.getOrDefault(num,0)+1);
        }
        for(int i:mp.keySet()){
            if(mp.get(i)==n){
                return i;
            }
        }
        return 0;
    }
}

File: Python/dictionary/N-Repeated Element in Size 2N Array.py
class Solution:
    def repeatedNTimes(self, nums: List[int]) -> int:
        sz = len(nums)
        n = sz // 2
        thisdict = {}

        for num in nums:
            thisdict[num] = thisdict.get(num, 0) + 1
        

        for k,v in thisdict.items():
            if v == n:
                return k
        return 0
